.github/workflows/macos.yaml

```diff
@@ -5,34 +5,72 @@ on: [push, pull_request]
 jobs:
 
   macos-build:
-    runs-on: ${{ matrix.macos-version }}
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        macos-version:
-          - 'macos-latest'
+        os:
```

Comment:
```suggestion
        os:
          - {arch: clang_64, os: macos-13}
          - {arch: arm64, os: macos-latest}
```

Reply:
This way you don't have to iterate twice on your matrix. 

{matrix.os.arch | matrix.os.os}